clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/MessagingKafkaConsumerConfigProperties.java

```diff
@@ -27,13 +27,12 @@
  * @see <a href="https://www.eclipse.org/hono/docs/api/event-kafka">Event API for Kafka Specification</a>
  * @see <a href="https://www.eclipse.org/hono/docs/api/command-and-control-kafka/">Command &amp; Control API for Kafka Specification</a>
  */
-// TODO rename to something that expresses that this is about the consumers used for *messaging*
-public class KafkaConsumerConfigProperties extends AbstractKafkaConsumerConfigProperties {
+public class MessagingKafkaConsumerConfigProperties extends AbstractKafkaConsumerConfigProperties {
 
     /**
      * Creates an instance.
      */
-    public KafkaConsumerConfigProperties() {
+    public MessagingKafkaConsumerConfigProperties() {
```

Comment:
I do not quite see how this class is specific to Hono's Messaging infrastructure or am I missing something?

Reply:
It enforces the configuration that is required by Hono's messaging APIs as defined in https://www.eclipse.org/hono/docs/api/kafka-api/. While it might be possible to use this configuration in other contexts, this is not always the case. For notifications, we could use JSON as the value type, while the messaging APIs require it to be a plain byte array.